boot.js

```diff
@@ -359,9 +360,24 @@ function thenify () {
     debug('thenify returning null because we are already booted')
     return
   }
+
+  // Calling resolve(this._server) would fetch the then
+  // property on the server, which will lead it here.
+  // If we do not break the recursion, we will loop
+  // forever.
+  if (this[kThenifyDoNotWrap]) {
+    this[kThenifyDoNotWrap] = false
+    return
```

Comment:
Should we return server here as well though? And also make it thenable

Reply:
The server is already thenable. The point of this check is to avoid it when passing the server to `resolve()` further down (see the comment).